### PR TITLE
21 hobbies profiles

### PR DESCRIPTION
**Pull request by Purrloin4**
## Description

Hobbies work with a list that you can select from

## Related Issue(s)

#closes 21 

## How Has This Been Tested?

later oops

## Checklist
Add check (later) if done, - if not applicable
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: vlinder/src/app/homepage/page.tsx [hunk 1]
```diff
@@ -1,31 +1,38 @@
 'use client';
 
-import React from 'react';
 import { createClient } from '@/utils/supabase/client'
-import { useState, useEffect, useCallback } from 'react';
+import React, { useState, useEffect, useCallback } from 'react';
+import { useRouter } from 'next/navigation';
 import { Button, Modal, ModalContent, ModalHeader, ModalBody, ModalFooter, Switch, Spacer, useDisclosure, Slider, CheckboxGroup, Checkbox } from '@nextui-org/react';
 
-interface Profile{
```
Comment (ThiewoutBekaert) [inline vlinder/src/app/homepage/page.tsx:8]: interface in a seperate file plzz :)) "vlinder\src\enums"

Comment (ThiewoutBekaert) [reply vlinder/src/app/homepage/page.tsx:8]: maby a diffrant folder for interfaces

File: vlinder/src/app/homepage/page.tsx [hunk 2]
```diff
@@ -1,31 +1,38 @@
...
+    disability: string;
+    profile_hobbies: ProfileHobby[];
 }
 
 
+
 export default function HomePage() {
     const supabase = createClient()
-    let userId: string;
-    if (process.env.NODE_ENV === 'test') {
```
Comment (ThiewoutBekaert) [inline vlinder/src/app/homepage/page.tsx:23]: maby add enum for env strings ('test')

File: vlinder/src/app/login/page.tsx [hunk 3]
```diff
@@ -5,30 +5,13 @@ import { Button, Input } from "@nextui-org/react";
 export default function LoginPage() {
   const widthClass = "w-[70%]"; 
   return (
-    <div className="flex items-center justify-center min-h-screen">
```
Comment (ThiewoutBekaert) [inline vlinder/src/app/login/page.tsx:8]: why is this replaced with shitty form inputs?

Comment (ThiewoutBekaert) [reply vlinder/src/app/login/page.tsx:8]: shity as in no layout

File: vlinder/src/app/myprofile/page.tsx [hunk 4]
```diff
@@ -3,9 +3,20 @@
 import React, { useState, useEffect } from 'react';
 import { createClient } from '@/utils/supabase/client';
 import { useRouter } from 'next/navigation';
+import { checkbox } from '@nextui-org/react';
 
 const supabase = createClient();
 
+interface Hobby {
```
Comment (ThiewoutBekaert) [inline vlinder/src/app/myprofile/page.tsx:10]: sepperate file

File: vlinder/src/app/myprofile/page.tsx [hunk 5]
```diff
@@ -232,15 +338,54 @@ export default function EditProfilePage() {
                     style={styles.textarea}
                 />
             </div>
-            <div style={styles.inputContainer}>
-                <label>Hobbies:</label>
+            <div>
+                <label>Hobbies:<br></br></label>
```
Comment (ThiewoutBekaert) [inline vlinder/src/app/myprofile/page.tsx:342]: br has no ending tag vs code fucks with it but you can use <br />

File: vlinder/src/app/myprofile/page.tsx [hunk 6]
```diff
@@ -232,15 +338,54 @@ export default function EditProfilePage() {
...
+                </label>
+            </div>
+        ))}
+    </div>
+</div>
+
+
             <div style={styles.checkboxContainer}>
                 <div
                     style={{
```
Comment (ThiewoutBekaert) [inline vlinder/src/app/myprofile/page.tsx:391]: a lot of inline styles ? i geuss the front end has to change this?

File: vlinder/src/app/profile/[id]/page.tsx [hunk 7]
```diff
@@ -6,6 +6,16 @@ import ProfileDetail from 'Components/ProfileDetail';
 
 const supabase = createClient();
 
+interface Hobby {
```
Comment (ThiewoutBekaert) [inline vlinder/src/app/profile/[id]/page.tsx:9]: seperate file

File: vlinder/tests/pages/homepage.test.js [hunk 8]
```diff
@@ -3,34 +3,40 @@ import { render, screen, fireEvent, waitFor, user } from '@testing-library/react
...
 
 describe('HomePage', () => {
-
     it('renders loading state initially', () => {
-        render(<HomePage />);
+        render(<HomePage/>);
         expect(screen.getByText('Loading...')).toBeInTheDocument();
     });
 
-    it('renders profiles after loading', async () => {
```
Comment (ThiewoutBekaert) [inline vlinder/tests/pages/homepage.test.js:20]: commants can be removed? for develpment?

**Review by ThiewoutBekaert — CHANGES_REQUESTED** [review]

File: vlinder/src/app/login/page.tsx (hunk 3 above)
Comment (Purrloin4) [reply vlinder/src/app/login/page.tsx:8]: as it is functional

File: vlinder/src/app/myprofile/page.tsx (hunk 4 above)
Comment (Purrloin4) [reply vlinder/src/app/myprofile/page.tsx:10]: no some are unique to file

File: vlinder/tests/pages/homepage.test.js (hunk 8 above)
Comment (Purrloin4) [reply vlinder/tests/pages/homepage.test.js:20]: no

**Review by ThiewoutBekaert — APPROVED** [review]

**Pull request merged**